spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -521,4 +539,67 @@ private DatastoreReaderWriter getDatastoreReadWriter() {
 						.currentTransactionStatus()).getTransaction()).getTransaction()
 				: this.datastore;
 	}
+
+	private <T> StructuredQuery exampleToQuery(Example<T> example, DatastoreQueryOptions queryOptions, boolean keyQuery) {
+		validateExample(example);
+
+		T probe = example.getProbe();
+		FullEntity.Builder<IncompleteKey> probeEntityBuilder = Entity.newBuilder();
+		this.datastoreEntityConverter.write(probe, probeEntityBuilder);
```

Comment:
Could use "convertToEntityForSave" method we have that wraps this the datastoreEntityConverter.write  method that we use elsewhere in this template

Reply:
We can't, because convertToEntityForSave recursively saves references and descendants in addition to conversion - we don't want to do that.